index.js

```diff
@@ -1,148 +1,148 @@
 };
 
-J2M.prototype.jira_to_html = function(str) {
-	return marked(this.to_markdown(str));
+J2M.prototype.jira_to_html = function (str) {
+  return marked(this.to_markdown(str));
 };
 
-J2M.prototype.to_markdown = function(str) {
-    return str
-        // Ordered Lists
-        .replace(/^[ \t]*(\*+)\s+/gm, function(match, stars) {
-            return Array(stars.length).join("  ") + '* ';
-        })
-        // Un-ordered lists
-        .replace(/^[ \t]*(#+)\s+/gm, function(match, nums) {
-            return Array(nums.length).join("  ") + '1. ';
-        })
-        // Headers 1-6
-        .replace(/^h([0-6])\.(.*)$/gm, function (match, level, content) {
-            return Array(parseInt(level) + 1).join('#') + content;
-        })
-        // Bold
-        .replace(/\*(\S.*)\*/g, '**$1**')
-        // Italic
-        .replace(/\_(\S.*)\_/g, '*$1*')
-        // Monospaced text
-        .replace(/\{\{([^}]+)\}\}/g, '`$1`')
-        // Citations (buggy)
-        //.replace(/\?\?((?:.[^?]|[^?].)+)\?\?/g, '<cite>$1</cite>')
-        // Inserts
-        .replace(/\+([^+]*)\+/g, '<ins>$1</ins>')
-        // Superscript
-        .replace(/\^([^^]*)\^/g, '<sup>$1</sup>')
-        // Subscript
-        .replace(/~([^~]*)~/g, '<sub>$1</sub>')
-        // Strikethrough
-        .replace(/\s+-(\S+.*?\S)-\s+/g, ' ~~$1~~ ')
-        // Code Block
-        .replace(/\{code(:([a-z]+))?([:|]?(title|borderStyle|borderColor|borderWidth|bgColor|titleBGColor)=.+?)*\}([^]*)\{code\}/gm, '```$2$5```')
-        // Pre-formatted text
-        .replace(/{noformat}/g, '```')
-        // Un-named Links
-        .replace(/\[([^|]+)\]/g, '<$1>')
-        // Named Links
-        .replace(/\[(.+?)\|(.+)\]/g, '[$1]($2)')
-        // Single Paragraph Blockquote
-        .replace(/^bq\.\s+/gm, '> ')
-        // Remove color: unsupported in md
-        .replace(/\{color:[^}]+\}([^]*)\{color\}/gm, '$1')
-        // panel into table
-        .replace(/\{panel:title=([^}]*)\}\n?([^]*?)\n?\{panel\}/gm, '\n| $1 |\n| --- |\n| $2 |')
-        // table header
-        .replace(/^[ \t]*((?:\|\|.*?)+\|\|)[ \t]*$/gm, function (match, headers) {
-            var singleBarred =  headers.replace(/\|\|/g,'|');
-            return '\n' + singleBarred + '\n' + singleBarred.replace(/\|[^|]+/g, '| --- ');
-        })
-        // remove leading-space of table headers and rows
-        .replace(/^[ \t]*\|/gm, '|');
+J2M.prototype.to_markdown = function (str) {
+  return str
+    // Ordered Lists
+    .replace(/^[ \t]*(\*+)\s+/gm, function (match, stars) {
+      return Array(stars.length).join("  ") + '* ';
+    })
+    // Un-ordered lists
+    .replace(/^[ \t]*(#+)\s+/gm, function (match, nums) {
+      return Array(nums.length).join("  ") + '1. ';
+    })
+    // Headers 1-6
+    .replace(/^h([0-6])\.(.*)$/gm, function (match, level, content) {
+      return Array(parseInt(level) + 1).join('#') + content;
+    })
+    // Bold
+    .replace(/(\s|^|\_)\*(\S.*)\*/g, '$1**$2**')

```

Comment:
so this regex will match sth like `phrase _with_underscore` and change it to `phrase **with**underscore`. should we add group 3 to catch the end of the special syntax?
eg: `replace(/(\s|^|\_)\_(\S.*)\_(\s|&|\_)/g, '$1**$2**$3')`

also, what do we want to do with `phrase _with_more_ underscores`? change to `phrase **with_more** underscores`?

Reply:
1) Yes absolutely, although I've done the implementation a little differently to allow for `Sentence end with bold *word*.` and similar cases. I've added unit tests to cover these scenarios. Good catch! 

2) Yes, that's what we want to do. The fix for (1) handles this and I've also added unit tests.